serving/resource_model_serving.go

```diff
@@ -0,0 +1,75 @@
+package serving
+
+import (
+	"context"
+	"time"
+
+	"github.com/databricks/databricks-sdk-go/retries"
+	"github.com/databricks/databricks-sdk-go/service/endpoints"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const DefaultProvisionTimeout = 45 * time.Minute
+
+func ResourceModelServing() *schema.Resource {
+	s := common.StructToSchema(
+		endpoints.CreateServingEndpoint{},
+		func(m map[string]*schema.Schema) map[string]*schema.Schema {
+			m["name"].ForceNew = true
+			return m
+		})
+
+	return common.Resource{
+		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			var e endpoints.CreateServingEndpoint
+			common.DataToStructPointer(d, s, &e)
+			endpoint, err := w.ServingEndpoints.CreateAndWait(ctx, e, retries.Timeout[endpoints.ServingEndpointDetailed](DefaultProvisionTimeout))
+			if err != nil {
+				return err
+			}
+			d.SetId(endpoint.Name)
+			return nil
+		},
+		Read: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			endpoint, err := w.ServingEndpoints.GetByName(ctx, d.Id())
+			if err != nil {
+				return err
+			}
+			return common.StructToData(*endpoint, s, d)
+		},
+		Update: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			var e endpoints.CreateServingEndpoint
```

Comment:
Curious why this uses a CreateServingEndpoint as opposed to an Update?

Reply:
Yeah there is no `UpdateServingEndpoint` model and that's because the [Update model function input](https://src.dev.databricks.com/databricks/databricks-sdk-go/-/blob/service/endpoints/api.go?L218)([EndpointCoreConfigInput](https://src.dev.databricks.com/databricks/databricks-sdk-go/-/blob/service/endpoints/model.go?L27)) is actually a field in CreateServingEndpoint, which as we mentioned [here](https://github.com/databricks/terraform-provider-databricks/pull/2054#discussion_r1127240376) is the overall schema we use for this resource.

Reply:
Also when we get the `ResourceData`, we get it in the same form as `CreateServingEndpoint` since that's how the resource will be written in HCL in the Terraform config, and we transform that into a populated `EndpointCoreConfigInput` which is why we re-specify the `EndpointCoreConfigInput.Name` field.